cilium/cmd/bpf_lb_list.go

```diff
@@ -66,7 +68,17 @@ func dumpSVC(serviceList map[string][]string) {
 
 		svcKey := key.(lbmap.ServiceKey)
 		svcVal := value.(lbmap.ServiceValue)
-		svc := svcKey.String()
+		svc := net.JoinHostPort(svcKey.GetAddress().String(), fmt.Sprintf("%d", svcKey.GetPort()))
```

Comment:
Could you explain why this change is needed, i.e. why calling `svcKey.String()` is not enough?

Reply:
That callback function parseSVCEntry receive the actual key and value by DumpParser()
```
lbmap.Service4MapV2.DumpWithCallbackIfExists
->m.DumpWithCallback
	->mk, mv, err = m.DumpParser(nextKey, value, mk, mv)
```
svcKey`s byte order has been changed before been return.
https://github.com/cilium/cilium/blob/master/pkg/maps/lbmap/ipv4.go#L59

This patch squash commits into one and also refer to your comment.

Reply:
Couldn't you do `svcKey.ToNetwork().String()` instead to DRY? Anyway, this looks hacky to me. One idea is to avoid changing endianness in the dump parsers and only do it in the `String()` methods. Otherwise, we are risking accidentally mixing endianness because it gets more difficult to track in what order structs are.

Reply:
Agree with you,
I`m preparing another patch that gets rid of ToNetwork() in DumpParser(); 
I plan to submit another PR to complete this optimization.